plugins/coreboot/fu-coreboot-fmap.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2019 9elements Agency GmbH <patrick.rudolph@9elements.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-device-metadata.h"
+#include "fu-device-private.h"
+#include "fu-plugin-coreboot.h"
+#include "fmap_serialized.h"
+
+/* Depends on the kernel module coreboot-fmap.ko */
+
+struct fmap *
+fu_plugin_coreboot_find_fmap (FuPlugin *plugin, GError **error)
+{
+	gchar *tmp = NULL;
+
+	g_debug ("coreboot: Trying /sys/firmware/fmap");
+
+	if (!g_file_get_contents ("/sys/firmware/fmap", &tmp, NULL, error))
+		return NULL;
+
+	g_debug ("coreboot: FMAP is available");
+
+	// FIXME: Do sanity check
+	return (struct fmap *)tmp;
+}
+
+gchar *
+fu_plugin_coreboot_find_cbfs_active_partition (FuPlugin *plugin, GError **error)
+{
+        gchar *tmp = NULL;
```

Comment:
g_autofree gchar

Reply:
the caller has to free it, added a comment